core/modules/filters/getvariablejson.js

```diff
@@ -0,0 +1,52 @@
+/*\
+title: $:/core/modules/filters/getvariablejson.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator to get widget variable info and
+Display as JSON with basic formatting
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Export our filter function
+*/
+exports.getvariablejson = function(source,operator,options) {
+	var results = [],
+		widget = options.widget;
+	// "replacer" must be defined, otherwise JSON.stringify will throw a circular reference error RSOD
+	var replacer= "params name value default text isFunctionDefinition isProcedureDefinition isWidgetDefinition isMacroDefinition isVariable".split(" ");
```

Comment:
I really don't think we should be exposing the `text` property. It is the same as the value retrieved from the variable in the usual way. Worse, it exposes an implementation detail of the caching mechanism.